send/insight-connector.pl

```diff
@@ -0,0 +1,255 @@
+
+### Example command - contains import token (t) and directory (d) with json objects to be imported.
+### All files in the directory will be imported to Jira Insight in a bulk.
+### If something fails, import is unfinished and is canceled with next calling of this script.
+=c
+./insight-connector.pl -d ../gen/spool/insight_mu -t {token}
+=cut
+#-----------------------------CONSTANTS------------------------------------
+# DEBUG can be turned on (1) or off (0)
+my $DEBUG=0;
+
+# Required setting for connection
+my $INFO_URL = 'https://api.atlassian.com/jsm/insight/v1/imports/info';
+my $GET_STATUS_URL;
+my $START_URL;
+my $SUBMIT_RESULTS_URL;
+my $GET_EXECUTION_STATUS;
+
+# Passed parameters
+my $token;
+my $importDirectory;
+
+# Other constants
+my $TYPE_GET = 'GET';
+my $TYPE_POST = 'POST';
+my $TYPE_DELETE = 'DELETE';
+my $STATUS_IDLE = 'IDLE';
+my $STATUS_RUNNING = 'RUNNING';
+# When finalChunk is sent to SUBMIT RESULTS endpoint, the import is finished
+my $finalChunk = '{"completed": true}';
+# How many times (+4s waiting) should the execution status be waited for to finish
+my $maxwait = 6;
+#--------------------------------------------------------------------------
+
+# Method with information about possible parameters of this script
+sub help {
+	return qq{
+Import all files (json) from import directory to Jira INSIGHT via REST API.
+Return help + exit 1 if help is needed.
+Return STDOUT + exit 0 if everything is ok.
+Return STDERR + exit >0 if error happens.
+---------------------------------------------------------
+Other options:
+ --help                | -h prints this help
+Mandatory parameters:
+ --token               | -t import token
+ --importDirectory     | -d command to call\n
+};
+}
+
+# Get options from input of script
+GetOptions("help|h"	=> sub {
+	print help;
+	exit 1;
+},
+	"token|t=s" => \$token,
+	"importDirectory|d=s" => \$importDirectory) || die help;
+
+if (!$token) {die "Token (-t) is required!\n"}
+if (!$importDirectory) {die "Import directory (-d) is required!\n"}
+
+# Call INFO endpoint with token to obtain endpoints
+my $info_links = callServer($INFO_URL, $TYPE_GET);
+if (!$info_links->{'links'}) {die "Missing data in server response!\n"}
+$START_URL = $info_links->{'links'}->{'start'};
+$GET_STATUS_URL = $info_links->{'links'}->{'getStatus'};
+
+# Start import
+checkImportReady();
+my $import_links = callServer($START_URL, $TYPE_POST);
+if (!$import_links->{'links'}) {die "Missing data in server response!\n"}
+$SUBMIT_RESULTS_URL = $import_links->{'links'}->{'submitResults'};
+$GET_EXECUTION_STATUS = $import_links->{'links'}->{'getExecutionStatus'};
+importFilesFromDirectory();
+
+# Finish import
+callServer($SUBMIT_RESULTS_URL, $TYPE_POST, $finalChunk);
+
+my $jsonResult;
+my $counter = 0;
+while ((!$jsonResult || !$jsonResult->{'status'} || $jsonResult->{'status'} ne "DONE") && $counter < $maxwait) {
+	if ($counter >= $maxwait) { die "Timeout for execution status, did not receive import status!\n" }
```

Comment:
This line should be processed after the while cycle.

Reply:
i moved the counter before the timeout check instead

Reply:
A little weird is that if the max wait would be 1, it would not even try at all. I would expect this behavior from the value "0".